internal/bminventory/inventory.go

```diff
@@ -2286,6 +2286,69 @@ func (b *bareMetalInventory) checkFileForDownload(ctx context.Context, clusterID
 	return nil
 }
 
+func (b *bareMetalInventory) GetHostIgnition(ctx context.Context, params installer.GetHostIgnitionParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	_, respBody, _, err := b.downloadHostIgnition(ctx, params.ClusterID.String(), params.HostID.String())
+	if err != nil {
+		log.WithError(err).Errorf("failed to download host %s ignition", params.HostID)
+		return common.GenerateErrorResponder(err)
+	}
+
+	respBytes, err := ioutil.ReadAll(respBody)
+	if err != nil {
+		log.WithError(err).Errorf("failed to read ignition content for host %s", params.HostID)
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	return installer.NewGetHostIgnitionOK().WithPayload(&models.HostIgnitionParams{Config: string(respBytes)})
+}
+
+func (b *bareMetalInventory) DownloadHostIgnition(ctx context.Context, params installer.DownloadHostIgnitionParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	fileName, respBody, contentLength, err := b.downloadHostIgnition(ctx, params.ClusterID.String(), params.HostID.String())
+	if err != nil {
+		log.WithError(err).Errorf("failed to download host %s ignition", params.HostID)
+		return common.GenerateErrorResponder(err)
+	}
+
+	return filemiddleware.NewResponder(installer.NewDownloadHostIgnitionOK().WithPayload(respBody), fileName, contentLength)
+}
+
+// downloadHostIgnition returns the ignition file name, the content as an io.ReadCloser, and the file content length
+func (b *bareMetalInventory) downloadHostIgnition(ctx context.Context, clusterID string, hostID string) (string, io.ReadCloser, int64, error) {
+	c, err := b.getCluster(ctx, clusterID, true)
+	if err != nil {
+		return "", nil, 0, err
+	}
+
+	// check if host id is in cluster
+	var host *models.Host
+	for _, h := range c.Hosts {
+		if h.ID.String() == hostID {
+			host = h
+		}
+	}
```

Comment:
Why not use getHost()?

Reply:
I need the cluster reference below to check the state, so `getCluster` brings back both for me in one db query.